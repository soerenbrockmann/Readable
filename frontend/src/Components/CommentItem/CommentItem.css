.commentitem {
  display: flex;
  flex-direction: column;
  border: 1px solid #ddd;
  padding: 10px;
  background-color: #f8f8f8;
}

.commentitemmain {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.commentitemmain > div {
  margin-right: 0.5em;
}

.commentitemmeta {
  font-style: italic;
}
