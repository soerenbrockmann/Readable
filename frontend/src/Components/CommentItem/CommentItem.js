import React from "react";
import PropTypes from "prop-types";
import "./CommentItem.css";
import Votes from "../Votes";
import FormatDate from "../FormatDate";

const CommentItem = props => {
  const { item, onDelete, onEdit, onUpVote, onDownVote } = props;
  const { timestamp, body, author, voteScore } = item;

  return (
    <div className="commentitem">
      <div>{body}</div>
      <div className="commentitemmain">
        <div className="meta">
          <div>{author} - </div>
          <FormatDate timestamp={timestamp} />
        </div>
        <div className="ondelete meta" onClick={onDelete}>
          Delete
        </div>
        <div className="ondelete meta" onClick={onEdit}>
          Edit
        </div>
      </div>

      <div>
        <Votes
          onUpVote={onUpVote}
          onDownVote={onDownVote}
          voteScore={voteScore}
        />
      </div>
    </div>
  );
};

Votes.propTypes = {
  onUpVote: PropTypes.func.isRequired,
  onDownVote: PropTypes.func.isRequired,
  voteScore: PropTypes.number
};

FormatDate.propTypes = {
  timestamp: PropTypes.number.isRequired
};

export default CommentItem;
