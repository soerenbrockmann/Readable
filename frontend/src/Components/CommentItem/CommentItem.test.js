import React from "react";
import { shallow } from "enzyme";
import CommentItem from "./CommentItem";

describe("<CommentItem />", () => {
  const onDelete = jest.fn();
  const onEdit = jest.fn();
  const onDownVote = jest.fn();
  const onUpVote = jest.fn();
  const item = {
    timestamp: 1468166872634,
    body: "",
    author: "",
    voteScore: 0
  };

  const commentItem = shallow(
    <CommentItem
      onDelete={onDelete}
      onEdit={onEdit}
      onDownVote={onDownVote}
      onUpVote={onUpVote}
      item={item}
    />
  );
  it("renders properly", () => {
    expect(commentItem).toMatchSnapshot();
  });
});
