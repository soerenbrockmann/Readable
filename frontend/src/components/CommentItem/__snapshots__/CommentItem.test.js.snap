// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentItem /> renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommentItem
    item={
        Object {
            "author": "",
            "body": "",
            "timestamp": 1468166872634,
            "voteScore": 0,
          }
    }
    onDelete={[Function]}
    onDownVote={[Function]}
    onEdit={[Function]}
    onUpVote={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div />,
        <div
          className="commentitemmain"
>
          <div
                    className="meta"
          >
                    <div>
                              
                               - 
                    </div>
                    <FormatDate
                              timestamp={1468166872634}
                    />
          </div>
          <div
                    className="ondelete meta"
                    onClick={[Function]}
          >
                    Delete
          </div>
          <div
                    className="ondelete meta"
                    onClick={[Function]}
          >
                    Edit
          </div>
</div>,
        <div>
          <Votes
                    onDownVote={[Function]}
                    onUpVote={[Function]}
                    voteScore={0}
          />
</div>,
      ],
      "className": "commentitem",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="meta"
>
              <div>
                            
                             - 
              </div>
              <FormatDate
                            timestamp={1468166872634}
              />
</div>,
            <div
              className="ondelete meta"
              onClick={[Function]}
>
              Delete
</div>,
            <div
              className="ondelete meta"
              onClick={[Function]}
>
              Edit
</div>,
          ],
          "className": "commentitemmain",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div>
                  
                   - 
</div>,
                <FormatDate
                  timestamp={1468166872634}
/>,
              ],
              "className": "meta",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "",
                    " - ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "",
                  " - ",
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "timestamp": 1468166872634,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Delete",
              "className": "ondelete meta",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Delete",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Edit",
              "className": "ondelete meta",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Edit",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Votes
            onDownVote={[Function]}
            onUpVote={[Function]}
            voteScore={0}
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onDownVote": [Function],
            "onUpVote": [Function],
            "voteScore": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div />,
          <div
            className="commentitemmain"
>
            <div
                        className="meta"
            >
                        <div>
                                    
                                     - 
                        </div>
                        <FormatDate
                                    timestamp={1468166872634}
                        />
            </div>
            <div
                        className="ondelete meta"
                        onClick={[Function]}
            >
                        Delete
            </div>
            <div
                        className="ondelete meta"
                        onClick={[Function]}
            >
                        Edit
            </div>
</div>,
          <div>
            <Votes
                        onDownVote={[Function]}
                        onUpVote={[Function]}
                        voteScore={0}
            />
</div>,
        ],
        "className": "commentitem",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="meta"
>
                <div>
                                
                                 - 
                </div>
                <FormatDate
                                timestamp={1468166872634}
                />
</div>,
              <div
                className="ondelete meta"
                onClick={[Function]}
>
                Delete
</div>,
              <div
                className="ondelete meta"
                onClick={[Function]}
>
                Edit
</div>,
            ],
            "className": "commentitemmain",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div>
                    
                     - 
</div>,
                  <FormatDate
                    timestamp={1468166872634}
/>,
                ],
                "className": "meta",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "",
                      " - ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "",
                    " - ",
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "timestamp": 1468166872634,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Delete",
                "className": "ondelete meta",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Delete",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Edit",
                "className": "ondelete meta",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Edit",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Votes
              onDownVote={[Function]}
              onUpVote={[Function]}
              voteScore={0}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onDownVote": [Function],
              "onUpVote": [Function],
              "voteScore": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
